engine.js

```diff
@@ -0,0 +1,96 @@
+"format cjs";
+
+var wrap = require('word-wrap');
+var cloneDeep = require('lodash.clonedeep');
+var map = require('lodash.map');
+var longest = require('longest');
+var rightPad = require('right-pad');
+
+// This can be any kind of SystemJS compatible module.
+// We use Commonjs here, but ES6 or AMD would do just
+// fine.
+module.exports = function (options) {
+
+  options = cloneDeep(options || {});
+  var types = options.types;
+
+  var length = longest(Object.keys(types)).length + 1;
+  var choices = map(types, function (type, key) {
```

Comment:
All our targets supports native array#map, right? Why use `lodash.map`?

Oh, it's because `types` is an object, not an array... I would personally prefer having it an array, but I guess I don't feel to strongly about that...

How about changing to the following to make it more clear what's happening, and to avoid to bring in lodash.map?

```js
var choices = Object.keys(types).map(function (key) {
  return {
    name: rightPad(key + ':', length) + ' ' + types[key].description,
    value: key
  }
})
```